src/Entity/Donation.php

```diff
@@ -242,4 +258,31 @@ public function getRetryPayload(): array
 
         return $payload;
     }
+
+    public function getSubscriptionEndedAt(): ?\DateTime
+    {
+        return $this->subscriptionEndedAt;
+    }
+
+    public function setSubscriptionEndedAt(?\DateTime $subscriptionEndedAt): void
+    {
+        $this->subscriptionEndedAt = $subscriptionEndedAt;
+    }
+
+    public function nextDonationAt(): \DateTime
```

Comment:
this method cannot work if donation is punctual

Reply:
Sure, I will throw an exception